src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt

```diff
@@ -2084,3 +2084,283 @@ Error({Kind:ErrorKind.InvalidArgument})
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), "yyyy-mm-dd;mm/dd/yyyy;mmmm dd yyyy")
+"1899-12-30"
+
+>> Text(DateTime(2023, 10, 26, 2, 53, 9, 0), "\T\:yyyy-mm-dd;\Z\:mm/dd/yyyy")
+"T:2023-10-26"
+
+>> Text(DateTime(1000, 2, 6, 7, 14, 25, 0), """T:""yyyy-mm-dd;""Z:""mm/dd/yyyy")
+"Z:02/06/1000"
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), """T:""yyyy-mm-dd;""Z:""mm/dd/yyyy")
+"T:1899-12-30"
+
+>> Text(DateTime(2023, 10, 26, 2, 53, 9, 0), "\T\:yyyy-mm-dd;\Z\:mm/dd/yyyy;\K\:mmmm dd yyyy")
+"T:2023-10-26"
+
+>> Text(DateTime(1000, 2, 6, 7, 14, 25, 0), "\T\:yyyy-mm-dd;\Z\:mm/dd/yyyy;\K\:mmmm dd yyyy")
+"Z:02/06/1000"
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), "\T\:yyyy-mm-dd;\Z\:mm/dd/yyyy;\K\:mmmm dd yyyy")
+"T:1899-12-30"
+
+>> Text(DateTime(2023, 10, 26, 2, 53, 9, 0), "yyyy-mm-dd\;mm/dd/yyyy")
+"2023-10-26;10/26/2023"
+
+>> Text(DateTime(1000, 2, 6, 7, 14, 25, 0), "yyyy-mm-dd\;mm/dd/yyyy")
+"1000-02-06;02/06/1000"
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), "yyyy-mm-dd\;mm/dd/yyyy")
+"1899-12-30;12/30/1899"
+
+>> Text(DateTime(2023, 10, 26, 2, 53, 9, 0), "yyyy-mm-dd\;mm/dd/yyyy\;mmmm dd yyyy")
+"2023-10-26;10/26/2023;October 26 2023"
+
+>> Text(DateTime(1000, 2, 6, 7, 14, 25, 0), "yyyy-mm-dd\;mm/dd/yyyy\;mmmm dd yyyy")
+"1000-02-06;02/06/1000;February 06 1000"
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), "yyyy-mm-dd\;mm/dd/yyyy\;mmmm dd yyyy")
+"1899-12-30;12/30/1899;December 30 1899"
+
+>> Text(DateTime(2023, 10, 26, 2, 53, 9, 0), "yyyy-mm-dd"";""mm/dd/yyyy"";""mmmm dd yyyy")
+"2023-10-26;10/26/2023;October 26 2023"
+
+>> Text(DateTime(1000, 2, 6, 7, 14, 25, 0), "yyyy-mm-dd"";""mm/dd/yyyy"";""mmmm dd yyyy")
+"1000-02-06;02/06/1000;February 06 1000"
+
+>> Text(DateTime(1899, 12, 30, 5, 12, 15, 0), "yyyy-mm-dd"";""mm/dd/yyyy"";""mmmm dd yyyy")
+"1899-12-30;12/30/1899;December 30 1899"
+
+>> Text(Time(14, 32, 25, 678), "hh:mm:ss.00")
+"14:32:25.67"
+
+>> Text(Time(-5, 23, 15, 356), "hh:mm:ss.00")
+"19:23:15.35"
+
+>> Text(Time(0, 0, 0, 0), "hh:mm:ss.00")
+"00:00:00.00"
+
+>> Text(Time(14, 32, 25, 678), "hh:mm:ss.00;hh/mm/ss")
+"14:32:25.67"
+
+>> Text(Time(-5, 23, 15, 356), "hh:mm:ss.00;hh/mm/ss")
+"19/23/15"
+
+>> Text(Time(0, 0, 0, 0), "hh:mm:ss.00;hh/mm/ss")
+"00:00:00.00"
+
+>> Text(Time(14, 32, 25, 678), "hh:mm:ss.00;hh/mm/ss;hh mm ss")
+"14:32:25.67"
+
+>> Text(Time(-5, 23, 15, 356), "hh:mm:ss.00;hh/mm/ss;hh mm ss")
+"19/23/15"
+
+>> Text(Time(0, 0, 0, 0), "hh:mm:ss.00;hh/mm/ss;hh mm ss")
+"00 00 0"
+
+>> Text(Time(14, 32, 25, 678), "\T\:hh:mm:ss.00;\Z\:hh/mm/ss")
+"T:14:32:25.67"
+
+>> Text(Time(-5, 23, 15, 356), """T:""hh:mm:ss.00;""Z:""hh/mm/ss")
+"Z:19/23/15"
+
+>> Text(Time(0, 0, 0, 0), """T:""hh:mm:ss.00;""Z:""hh/mm/ss")
+"T:00:00:00.00"
+
+>> Text(Time(14, 32, 25, 678), "\T\:hh:mm:ss.00;\Z\:hh/mm/ss;\K\:hh mm ss")
+"T:14:32:25.67"
+
+>> Text(Time(-5, 23, 15, 356), "\T\:hh:mm:ss.00;\Z\:hh/mm/ss;\K\:hh mm ss")
+"\Z\:19/23/15"
+
+>> Text(Time(0, 0, 0, 0), "\T\:hh:mm:ss.00;\Z\:hh/mm/ss;\K\:hh mm ss")
+"\K\:00 00 0"
```

Comment:
This should be "K:00 00 00" : we have 2 's', and the characters K and : are escaped

Reply:
fix logic and update test